### PR TITLE
LibCrypto: Add PBKDF2

### PR DESCRIPTION
**Pull request by stelar7**

### CODE REVIEW

**Review by BenWiederhake — APPROVED** [review]
I love how clear this is, and now I really get the advantage of inline spec-links. Awesome :D

File: Userland/Libraries/LibCrypto/Hash/PBKDF2.h [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    {
+        PRF prf(password);
+
+        // Note: hLen denotes the length in octets of the pseudorandom function output
+        u32 h_len = prf.digest_size();
+
+        // 1. If dkLen > (2^32 - 1) * hLen, output "derived key too long" and stop.
+        if (key_length_bytes > (AK::pow(2.0, 32.0) - 1) * h_len) {
+            return Error::from_string_view("derived key too long"sv);
+        }
```
Comment (GMTA) [inline Userland/Libraries/LibCrypto/Hash/PBKDF2.h:31]: You can leave out the braces for single line `if` blocks

File: Userland/Libraries/LibCrypto/Hash/PBKDF2.h [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+                r_temp.export_data(ti.bytes());
+            }
+
+            //  4. Concatenate the blocks and extract the first dkLen octets to produce a derived key DK:
+            if (i == l) {
+                key.overwrite((i - 1) * h_len, ti.data(), r);
+            } else {
+                key.overwrite((i - 1) * h_len, ti.data(), h_len);
+            }
```
Comment (GMTA) [inline Userland/Libraries/LibCrypto/Hash/PBKDF2.h:81]:
```suggestion
            key.overwrite((i - 1) * h_len, ti.data(), i == l ? r : h_len);
```

**Review by GMTA — CHANGES_REQUESTED** [review]
Nice!

**Review by GMTA — APPROVED** [review]

**Pull request merged**